components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/internal/util/WSO2PayloadUtils.java

```diff
@@ -589,4 +590,51 @@ public static boolean isUserBasedGrantType(String grantType) {
 
         return grantType != null && USER_BASED_GRANT_TYPES.contains(grantType);
     }
+
+    /**
+     * Retrieves the flow action from the Flow object.
+     *
+     * @param flow The Flow object.
+     * @return The name of the flow action or null if the flow is null.
+     */
+    public static String getFlowAction(Flow flow) {
```

Comment:
When we think of getting the Action, it is simply checking the flow in the identity context and setting it. 

I think we always pass the flow in the Identity Context to these methods. So it would be cleaner if we do that part inside this method too.

anyway we can do this as an improvement